firmware/firmware.ino

```diff
@@ -191,45 +191,47 @@ void setup()
   // Start the system in the disconnected state
   system_state = DISCONNECTED;
   
-  // TODO: replace this with smart hooks.
-  // Initialize sensors
-  platypus::sensors[0] = new platypus::GY26Compass(0);
-  platypus::sensors[1] = new platypus::GY26Compass(1);
-  platypus::sensors[2] = new platypus::GY26Compass(2);
-  platypus::sensors[3] = new platypus::ServoSensor(3);
+  // Set ADC Precision:
+  analogReadResolution(12);
+
+  // Initialize and power all peripherals (WiFi & Pump)
+  platypus::peripherals[0] = new platypus::Peripheral(0, true);
+  platypus::peripherals[1] = new platypus::Peripheral(1, true);
+
+  // Initialize External sensors
+  platypus::sensors[0] = new platypus::ES2(0, 0);
+  platypus::sensors[1] = new platypus::AtlasPH(1, 1);
+  platypus::sensors[2] = new platypus::EmptySensor(2, 2);
+  platypus::sensors[3] = new platypus::ServoSensor(3, 3);
+
+  // Initialize Internal sensors
+  platypus::sensors[4] = new platypus::BatterySensor(4);
+  platypus::sensors[5] = new platypus::IMU(5);
   
   // Initialize motors
   platypus::motors[0] = new platypus::Dynamite(0);
   platypus::motors[1] = new platypus::Dynamite(1);
 
-  // Power all peripherals
-  platypus::motors[0]->enablePower(true);
-  platypus::motors[1]->enablePower(true);
-
   // Make the ADK buffers into null terminated string.
   debug_buffer[INPUT_BUFFER_SIZE] = '\0';
   input_buffer[INPUT_BUFFER_SIZE] = '\0';
   output_buffer[OUTPUT_BUFFER_SIZE] = '\0';
-
-  // Set ADC Precision:
-  analogReadResolution(12);
   
   // Create secondary tasks for system.
   Scheduler.startLoop(motorUpdateLoop);
   Scheduler.startLoop(serialConsoleLoop);
-  Scheduler.startLoop(batteryUpdateLoop);
 
   // Initialize Platypus library.
   platypus::init();
   
   // Print header indicating that board successfully initialized
-  /*Serial.println(F("------------------------------"));
+  Serial.println(F("------------------------------"));
```

Comment:
This is what we currently have in the firmware:
char applicationName[] = "Platypus Server"; // the app on Android
char accessoryName[] = "Platypus Control Board"; // your Arduino board
char companyName[] = "Platypus LLC";
char versionNumber[] = "3.0";
char serialNumber[] = "3";
char url[] = "http://senseplatypus.com";

I remember updating the version number broke the phone's ability to communicate with the eboard.